addons/website_hr_recruitment/controllers/main.py

```diff
@@ -279,12 +281,77 @@ def jobs_compatibility(self, country=None, department=None, office_id=None, cont
+        message_input = ''
+        if 'name' in kwargs:
+            domain = expression.OR([domain, [('partner_name', '=ilike', kwargs['name'])]])
+            message_input = kwargs['name']
+        if 'email' in kwargs:
+            domain = expression.OR([domain, [('email_normalized', '=', email_normalize(kwargs['email']))]])
+            message_input = kwargs['email']
+        if 'phone' in kwargs:
+            domain = expression.OR([domain, ['|', ('partner_phone', '=', kwargs['phone']),
+                                             ('partner_mobile', '=', kwargs['phone'])]])
+            message_input = kwargs['phone']
+        if 'linkedin' in kwargs:
+            domain = expression.OR([domain, [('linkedin_profile', '=ilike', kwargs['linkedin'])]])
+            message_input = kwargs['linkedin']
+        return domain, message_input
+
+    def _get_ongoing_application_response(self, ongoing_application, job_id, message_input):
+        message = _("An application already exists for %s. Duplicates might be rejected.", message_input)
+        if ongoing_application.user_id:
+            message += _(" In case of issue, contact %s", ongoing_application.user_id.name)
+            if ongoing_application.user_id.email:
+                message += _(", %s", ongoing_application.user_id.email)
+            if ongoing_application.user_id.phone:
+                message += _(", %s", ongoing_application.user_id.phone)
+        return {
+            'applied_same_job': ongoing_application.job_id.id == int(job_id),
+            'applied_other_job': True,
+            'message': message
+        }
+
+    def _get_refused_application_response(self, job_id):
+        message = _("You applied for this position less than 6 months ago, and have been rejected. Please don't reapply unless you have a good reason.")
+        return {
+            'applied_same_job': True,
+            'applied_other_job': False,
+            'message': message
+        }
+
     @http.route('/website_hr_recruitment/check_recent_application', type='json', auth="public")
-    def check_recent_application(self, email, job_id):
-        date_limit = datetime.now() - timedelta(days=90)
-        domain = [('email_from', '=ilike', email),
-                  ('create_date', '>=', date_limit)]
-        recent_applications = http.request.env['hr.applicant'].sudo().search(domain)
-        response = {'applied_same_job': any(a.job_id.id == int(job_id) for a in recent_applications),
-                    'applied_other_job': bool(recent_applications)}
-        return response
+    def check_recent_application(self, job_id, **kwargs):
+        Applicant = http.request.env['hr.applicant'].sudo()
+        search_domain, message_input = self._build_search_domain(kwargs)
+
+        # No search criteria provided
+        if not search_domain:
+            return {
+                'applied_same_job': False,
+                'applied_other_job': False,
+                'message': ""
```

Comment:
nitpicking :slightly_smiling_face: (same below)
```suggestion
                'message': '',
```